Sources/Client/Client/API/Client+Devices.swift

```diff
@@ -0,0 +1,139 @@
+//
+//  Client+Devices.swift
+//  StreamChatCore
+//
+//  Created by Alexey Bukhtin on 09/01/2020.
+//  Copyright © 2020 Stream.io Inc. All rights reserved.
+//
+
+import Foundation
+
+// MARK: - Devices
+
+public extension Client {
+    
+    /// Add a device for Push Notifications.
+    /// - Parameters:
+    ///   - deviceToken: a device token.
+    ///   - completion: an empty completion block.
+    @discardableResult
+    func addDevice(deviceToken: Data, _ completion: @escaping Client.Completion<EmptyData> = { _ in }) -> URLSessionTask {
+        if deviceToken.isEmpty {
+            completion(.failure(.emptyDeviceToken))
+            return .empty
+        }
+        
+        return addDevice(deviceId: deviceToken.deviceToken, completion)
+    }
+    
+    /// Add a device for Push Notifications.
+    /// - Parameters:
+    ///   - deviceId: a Push Notifications device identifier.
+    ///   - completion: an empty completion block.
+    @discardableResult
+    func addDevice(deviceId: String, _ completion: @escaping Client.Completion<EmptyData> = { _ in }) -> URLSessionTask {
+        guard let currentUser = user else {
```

Comment:
why don't we move this check in `request`? AFAICT it does not make sense to do any API call (except for `/guest`) before the client has a user set

Reply:
this way we don't have to add guard in a thousands places

Reply:
To make a request I need a `userId`.